control_plotter.py

```diff
@@ -151,24 +154,28 @@
 	htmp = "h_tmp"
 	#h_sig = ROOT.TH1D(htmp, htmp, *plot['binning'])
 	#signal_dict["chain"][0].Draw(plot['var']+">>%s"%htmp, signal_dict['weight']+"*("+plot_cut+")", 'goff')
-	h_sig = getPlotFromChain(signal_dict["chain"][0], plot['var'], plot['bin'], cutString = plot_cut, weight = signal_dict["weight"] ,addOverFlowBin='both',variableBinning=plot["bin_set"])
-	h_sig.SetLineColor(signal_dict["color"])
-	h_sig.SetLineWidth(3)
-	h_sig.GetXaxis().SetNdivisions(505)
-	h_sig.GetYaxis().SetTitle(plot['y_axis'])
-	h_sig.SetTitle("")
-	h_sig.Draw("Histo Same")
-	leg_sig.AddEntry(h_sig, signal_dict['tex'],"l")
-	print("Integral of BKG:" , stack_hist.Integral())   
-	print("Integral of Signal:" , h_sig.Integral()) 
-	leg.SetFillColor(0)
-	leg.SetLineColor(0)
-	leg.Draw()
-	leg_sig.SetFillColor(0)
-	leg_sig.SetLineColor(0)
-	leg_sig.Draw()
+	if not plot_sig_stack :
+		h_sig = getPlotFromChain(signal_dict["chain"][0], plot['var'], plot['bin'], cutString = plot_cut, weight = signal_dict["weight"] ,addOverFlowBin='both',variableBinning=plot["bin_set"])
+		h_sig.SetFillColor(signal_dict["color"])
+		h_sig.SetLineColor(ROOT.kBlack)
+		h_sig.SetLineWidth(1)
+		h_sig.GetXaxis().SetNdivisions(505)
+		h_sig.GetYaxis().SetTitle(plot['y_axis'])
+		h_sig.SetTitle("")
+		h_sig.Draw("Histo Same")
+		leg_sig.AddEntry(h_sig, signal_dict['tex'],"l")
```

Comment:
Here when you put legends to indent you are telling that draw the legend only when the signal is not stack.